packages/design-system-react/src/components/icon/Icon.test.tsx

```diff
@@ -88,10 +88,17 @@ describe('Icon error cases', () => {
   });
 
   it('should warn and return null when name prop is missing', () => {
-    const { container } = render(
-      <Icon {...({ name: undefined } as unknown as IconProps)} />,
-    );
+    // @ts-expect-error Testing undefined name prop
+    const { container } = render(<Icon {...({} as Partial<IconProps>)} />);
+
     expect(consoleSpy).toHaveBeenCalledWith('Icon name is required');
     expect(container.firstChild).toBeNull();
   });
+
+  it('should warn and return null when icon is not found', () => {
+    const { container } = render(<Icon name={'NonExistentIcon' as IconName} />);
+
+    expect(consoleSpy).toHaveBeenCalledWith('Icon "NonExistentIcon" not found');
+    expect(container.firstChild).toBeNull();
+  });
 });
```

Comment:
Adding error handling tests for the Icon component:
1. Tests the case when name prop is missing:
- Uses @ts-expect-error to intentionally pass undefined name
- Verifies warning message is shown
- Confirms component returns null

2. Tests the case when an invalid icon name is provided:
- Attempts to render non-existent icon
- Verifies appropriate warning message
- Confirms component returns null

These tests ensure our Icon component fails gracefully with helpful warning messages instead of throwing errors.